### PR TITLE
fix: second evaluation would have cache

### PR DESCRIPTION
**Pull request by rebelchris**
### Describe what this PR does
- We had a small issue with onboarding if you where in the experiment, then complete onboarding the `shouldEvaluate` becomes false, but it would not retrigger unless you did a hard refresh on the page.
- This introduces a second trigger to always return false when already evaluated and this value changes

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-{number} #done

### Preview domain
https://fix-second-eval-fix.preview.app.daily.dev

### CODE REVIEW

File: packages/shared/src/hooks/useConditionalFeature.ts [hunk 1]
```diff
@@ -24,11 +24,19 @@ export const useConditionalFeature = <T extends JSONValue>({
...
+    feature.id,
+    shouldEvaluate,
+  );
 
   const { data: featureValue, isLoading } = useQuery(
     queryKey,
     () => {
+      if (!shouldEvaluate) {
+        return false as WidenPrimitives<T>;
+      }
```
Comment (capJavert) [inline packages/shared/src/hooks/useConditionalFeature.ts:39]: What about features that are not bool?

Comment (rebelchris) [reply packages/shared/src/hooks/useConditionalFeature.ts:39]: I can change it to the default value, would that make more sense I guess?

Comment (capJavert) [reply packages/shared/src/hooks/useConditionalFeature.ts]: Yes, just to support other feature flags in this hook.

**Review by capJavert — APPROVED** [review]

File: packages/shared/src/hooks/useConditionalFeature.ts (hunk 1 above)
Comment (rebelchris) [reply packages/shared/src/hooks/useConditionalFeature.ts]: @capJavert Although, at some point we might change default to true, and then you get this experiment forever.
But guess once we hit that we can add something like `valueWhenNotEvaluated` or something.

Comment (capJavert) [reply packages/shared/src/hooks/useConditionalFeature.ts]: Hmm, don't know at top of my head, but sounds complicated and yes something to evaluate for future usage of the hook.

Comment (capJavert) [reply packages/shared/src/hooks/useConditionalFeature.ts]: Basically just returning something that GB does not return feels wrong right? We should hide the added UI in regards to something else and feature value should always be true if user is part of experiment.

Comment (rebelchris) [reply packages/shared/src/hooks/useConditionalFeature.ts]: Yeah correct, for this flow it's fine, but can imagine there might be something where it won't work ilke this.
There we could potentially use the initial evaluation for rendering determination.

**Pull request merged**